nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateRecordStats.java

```diff
@@ -133,104 +141,106 @@ public Set<Relationship> getRelationships() {
 
     @Override
     public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
-        FlowFile input = session.get();
-        if (input == null) {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
             return;
         }
 
         try {
-            Map<String, RecordPath> paths = getRecordPaths(context, input);
-            Map<String, String> stats = getStats(input, paths, context, session);
-
-            input = session.putAllAttributes(input, stats);
-
-            session.transfer(input, REL_SUCCESS);
-
-        } catch (Exception ex) {
-            getLogger().error("Error processing stats.", ex);
-            session.transfer(input, REL_FAILURE);
+            final Map<String, RecordPath> recordPath = getRecordPaths(context, flowFile);
```

Comment:
Did you use the singular form here on purpose or accidentally? I'd prefer `recordPaths` over `recordPath`.

Reply:
That's worth adjusting @EndzeitBegins, thanks.